glean-core/ios/Glean/Metrics/DatetimeMetric.swift

```diff
@@ -0,0 +1,144 @@
+                Glean.shared.handle,
+                self.handle,
+                Int32(components.year ?? 0),
+                UInt32(components.month ?? 0),
+                UInt32(components.day ?? 0),
+                UInt32(components.hour ?? 0),
+                UInt32(components.minute ?? 0),
+                UInt32(components.second ?? 0),
+                Int64(components.nanosecond ?? 0),
+                Int32(components.timeZone!.secondsFromGMT(for: components.date!))
+            )
+        }
+    }
+
+    /// Tests whether a value is stored for the metric for testing purposes only. This function will
+    /// attempt to await the last task (if any) writing to the the metric's storage engine before
+    /// returning a value.
+    ///
+    /// - parameters:
+    ///     * pingName: represents the name of the ping to retrieve the metric for.
+    ///                 Defaults to the first value in `sendInPings`.
+    /// - returns: true if metric value exists, otherwise false
+    func testHasValue(_ pingName: String? = nil) -> Bool {
+        Dispatchers.shared.assertInTestingMode()
+
+        let pingName = pingName ?? self.sendInPings[0]
+        return glean_datetime_test_has_value(Glean.shared.handle, self.handle, pingName) != 0
+    }
+
+    /// Returns the string representation of the stored value for testing purposes only. This function
+    /// will attempt to await the last task (if any) writing to the the metric's storage engine before returning
+    ///  a value.
+    ///
+    /// Throws a `Missing value` exception if no value is stored
+    ///
+    /// - parameters:
+    ///     * pingName: represents the name of the ping to retrieve the metric for.
+    ///                 Defaults to the first value in `sendInPings`.
+    ///
+    /// - returns:  value of the stored metric
+    func testGetValueAsString(_ pingName: String? = nil) throws -> String {
+        Dispatchers.shared.assertInTestingMode()
+
+        let pingName = pingName ?? self.sendInPings[0]
+
+        if !testHasValue(pingName) {
+            throw "Missing value"
+        }
+
+        return String(
+            freeingRustString: glean_datetime_test_get_value_as_string(
+                Glean.shared.handle,
+                self.handle,
+                pingName
+            )
+        )
+    }
+
+    /// Returns the stored value for testing purposes only. This function will attempt to await the
+    /// last task (if any) writing to the the metric's storage engine before returning a value.
+    ///
+    /// Throws a `Missing value` exception if no value is stored
+    ///
+    /// - parameters:
+    ///     * pingName: represents the name of the ping to retrieve the metric for.
+    ///                 Defaults to the first value in `sendInPings`.
+    ///
+    /// - returns:  value of the stored metric
+    func testGetValue(_: String? = nil) throws -> Date {
+        return Date.fromISO8901String(dateString: try testGetValueAsString())!
```

Comment:
Is `pingName` being passed to `testGetValueAsString` here? If not, then maybe we should add a test case for this.

Reply:
iso8901 -> iso8601?

Reply:
> Is `pingName` being passed to `testGetValueAsString` here? If not, then maybe we should add a test case for this.

Good catch, it wasn't being passed through.



> iso8901 -> iso8601?

Oops..  good eye!